pkg/api/controller.go

```diff
@@ -2520,7 +2520,7 @@ func (c *Controller) Commit(w http.ResponseWriter, r *http.Request, body apigen.
 		metadata = body.Metadata.AdditionalProperties
 	}
 
-	newCommit, err := c.Catalog.Commit(ctx, repository, branch, body.Message, user.Committer(), metadata, body.Date, params.SourceMetarange, graveler.WithForce(swag.BoolValue(body.Force)))
+	newCommit, err := c.Catalog.Commit(ctx, repository, branch, body.Message, user.Committer(), metadata, body.Date, params.SourceMetarange, body.AllowEmpty, graveler.WithForce(swag.BoolValue(body.Force)))
```

Comment:
use swag.BoolValue to pass allow empty to the underlying layer. we no longer need pointer to represent optional.